core/server/adapters/scheduling/post-scheduling/index.js

```diff
@@ -1,24 +1,79 @@
-const Promise = require('bluebird'),
-    moment = require('moment'),
-    localUtils = require('../utils'),
-    common = require('../../../lib/common'),
-    models = require('../../../models'),
-    urlUtils = require('../../../lib/url-utils'),
-    _private = {};
+const Promise = require('bluebird');
+const moment = require('moment');
+const jwt = require('jsonwebtoken');
+const localUtils = require('../utils');
+const common = require('../../../lib/common');
+const models = require('../../../models');
+const urlUtils = require('../../../lib/url-utils');
+const _private = {};
+const SCHEDULED_RESOURCES = ['post', 'page'];
+
+/**
+ * @description Load the internal scheduler integration
+ *
+ * @return {Promise}
+ */
+_private.getSchedulerIntegration = function () {
+    return models.Integration.findOne({slug: 'ghost-scheduler'}, {withRelated: 'api_keys'})
+        .then((integration) => {
+            if (!integration) {
+                throw new common.errors.NotFoundError({
+                    message: common.i18n.t('errors.api.resource.resourceNotFound', {
+                        resource: 'Integration'
+                    })
+                });
+            }
+            return integration.toJSON();
+        });
+};
+
+/**
+ * @description Get signed admin token for making authenticated scheduling requests
+ *
+ * @return {Promise}
+ */
+_private.getSignedAdminToken = function (options) {
+    const {model, apiUrl, integration} = options;
+    let key = integration.api_keys[0];
+
+    const JWT_OPTIONS = {
+        keyid: key.id,
+        algorithm: 'HS256',
+        audience: apiUrl
+    };
+
+    // Default token expiry is till 10 minutes after scheduled time
+    // or if published_at is in past then till 10 minutes after blog start
+    // and never before 10 mins to publish time
+    let tokenExpiry = moment(model.get('published_at')).add(10, 'm');
+    if (tokenExpiry.isBefore(moment())) {
+        tokenExpiry = moment().add(10, 'm');
+    }
+
+    return jwt.sign(
+        {
+            exp: tokenExpiry.unix(),
+            nbf: moment(model.get('published_at')).subtract(10, 'm').unix()
```

Comment:
Think we could somehow reuse `offsetInMinutes` [from SchedulingDefault](https://github.com/TryGhost/Ghost/blob/2.31.0/core/server/adapters/scheduling/SchedulingDefault.js#L26) here? That's the only thing that I could think of when pinning the nbf to this interval

Reply:
🤔 I think the `SchedulingDefault` values should stay unrelated as that defines behavior for default scheduler adapter while this is about adjusting token validity for all generated URLs irrespective of default adapter ? Not sure though tbh if/how they should be linked